project-oak/oak
# Encapsulate Remote Attestation generation and verification

- various tech debt repayments throughout the codebase
- create separate crate with placeholder implementation of AMD-specific remote attestation functionality
- remove explicit TEE certificate argument that was previously passed everywhere; this functionality (and corresponding root cert) is now expected to be implemented in the remote attestation verification trait (the generation will be done in hardware by the generator trait); e.g. an Intel verifier will have the Intel root cert baked in, while an AMD verifier will have the AMD root cert baked in.
- clarify terminology `attestation_info` vs `attestation_report`
- remove ConfigurationInfo verifier; this was not doing anything anyways, and will need to be designed correctly as part of #2911 
- fix documentation throughout (mainly links and formatting)
- use generic types to determine the remote attestation generation and verification (so that they can be reified and optimized at compile time)
- correctly account for ephemeral public key, signing key, and additional data in attestation report (previously this was inconsistent and possibly incorrect)
- use a proper way of hashing multiple values of different lenghts (by hashing individual values first)
- make client and server attestation more symmetric (though there are still asymmetries, see #2914)
- Fix #2910
- Fix #2909
- Fix #2919
